.github/workflows/e2e-nvidia-l4-x1.yml

```diff
@@ -135,7 +135,7 @@ jobs:
           # https://github.com/instructlab/instructlab/issues/1821
           # install with Torch and build dependencies installed
           python3.11 -m pip install -v packaging wheel setuptools-scm
-          python3.11 -m pip install -v .[cuda] -r requirements-vllm-cuda.txt
+          python3.11 -m pip install -v .[cuda] -r requirements-vllm-cuda.txt -c constraints.txt
```

Comment:
This change will never trigger in a PR build, and therefore, the medium E2E job will always fail.

Reply:
There should probably be a way to test these changes against e2e without first merging them...

Reply:
We can do this with the large and XL e2e jobs because they already have `workflow_dispatch` defined under the trigger conditions. This medium e2e workflow does not have `workflow_dispatch` defined under the `release-v0.24` branch. So, when I try to add `workflow_dispatch` in my PR, I can't trigger these changes manually. (I think this is a security feature so that someone doesn't print out secrets, etc.)

This is exactly what the medium e2e job sees no matter what:

<img width="864" alt="Screenshot 2025-02-24 at 9 14 49 AM" src="https://github.com/user-attachments/assets/2ce7c55e-75f0-4f58-8b21-d7c3298d3175" />

Reply:
Is [this](https://github.com/instructlab/instructlab/actions/runs/13501409525/job/37721057060?pr=3176) the failure that you attribute to this issue?

```
RuntimeError: Failed to import transformers.modeling_utils because of the following error (look up to see its traceback):
/actions-runner/_work/instructlab/instructlab/venv/lib64/python3.11/site-packages/flash_attn_2_cuda.cpython-311-x86_64-linux-gnu.so: undefined symbol: _ZNK3c1011StorageImpl27throw_data_ptr_access_errorEv
+ rm -rf /home/tmp/tmp.1MEOpG6zQc
```

Reply:
Yes. That error is caused by the fact `torch-2.6.0` is installed in the medium E2E job (i.e., `constraints.txt` is not being passed in)

Essentially, the error is caused by a mismatch between vLLM and PyTorch versioning. We have `vllm==7.2.0`, which only works with `torch>=2.5.0,<2.6.0`. Since `torch==2.6.0` is installed, we get that error unless we upgrade `vllm`.